tests/Server.php

```diff
@@ -140,6 +140,13 @@ public static function stop()
         if (self::$started) {
             self::getClient()->request('DELETE', 'guzzle-server');
         }
+        $tries = 0;
+        while (self::isListening() && ++$tries < 5) {
+            usleep(100000);
+        }
+        if (self::isListening()) {
+            throw new \RuntimeException('Unable to stop node.js server');
+        }
```

Comment:
I started having some weird things happen at some point, not related to the PR sorry!

I think I had a leftover process, so I added some more checks here to make sure we successfully kill the test server.